src/modules/customers/customers.service.ts

```diff
@@ -22,4 +37,120 @@ export class CustomersService {
       }),
     );
   }
+
+  async getOrCreateCustomer(res: Response, DTFSCustomerDto: DTFSCustomerDto): Promise<GetCustomersResponse> {
```

Comment:
This function is quite big, can we split it to smaller helpers to avoid too many nested `if` and `try` statements